tests/testsuite/install.rs

```diff
@@ -2507,3 +2508,112 @@ fn install_incompat_msrv() {
 ")
         .with_status(101).run();
 }
+
+fn v1_path() -> PathBuf {
+    cargo_home().join(".crates.toml")
+}
+
+fn v2_path() -> PathBuf {
+    cargo_home().join(".crates2.json")
+}
+
+fn load_crates1() -> toml::Value {
+    toml::from_str(&fs::read_to_string(v1_path()).unwrap()).unwrap()
+}
+
+fn load_crates2() -> serde_json::Value {
+    serde_json::from_str(&fs::read_to_string(v2_path()).unwrap()).unwrap()
+}
```

Comment:
nit: Could we inline these functions in `assert_tracker_noexistence`? They doesn't seem to be shared elsewhere and scattering a single logic to multiple places make it a bit harder to read.